File: lib/config/generic_camera_configuration.dart
enum AssetType { photo, video }

class CameraPosition {
  static const int back = 1;
  static const int front = 2;
}

class FlashMode {
  static const int off = 0;
  static const int on = 1;
  static const int auto = 2;
}

class TorchMode {
  static const int off = 0;
  static const int on = 1;
  static const int auto = 2;
}

class GenericCameraConfiguration {
  AssetType captureMode;
  bool canCaptureMultiplePhotos;
  int cameraPosition;
  int cameraPhotoFlash;
  int cameraVideoTorch;

  GenericCameraConfiguration({
    this.captureMode = AssetType.photo,
    this.canCaptureMultiplePhotos = false,
    this.cameraPosition = CameraPosition.back,
    this.cameraPhotoFlash = FlashMode.auto,
    this.cameraVideoTorch = TorchMode.auto,
  });

  Map<String, dynamic> toJson() {
    return {
      'captureMode': captureMode.index,
      'canCaptureMultiplePhotos': canCaptureMultiplePhotos,
      'cameraPosition': cameraPosition,
      'cameraPhotoFlash': cameraPhotoFlash,
      'cameraVideoTorch': cameraVideoTorch,
    };
  }
}

File: android/src/main/java/com/example/flutter_generic_camera/model/GenericCameraConfiguration.kt
package com.example.flutter_generic_camera.model

import com.example.flutter_generic_camera.enum.AssetType
import com.example.flutter_generic_camera.enum.CameraPosition
import com.example.flutter_generic_camera.enum.FlashMode
import com.example.flutter_generic_camera.enum.TorchMode

data class GenericCameraConfiguration(
    var captureMode: AssetType /*= AssetType.PHOTO*/,
    var canCaptureMultiplePhotos: Boolean /*= false*/,
    var cameraPosition: Int /*= CameraPosition.BACK*/,
    var cameraPhotoFlash: Int /*= FlashMode.AUTO*/,
    var cameraVideoTorch: Int /*= TorchMode.AUTO*/
) : java.io.Serializable {
    fun toJson(): Map<String, Any> {
        return mapOf(
            "captureMode" to captureMode.ordinal,
            "canCaptureMultiplePhotos" to canCaptureMultiplePhotos,
            "cameraPosition" to cameraPosition,
            "cameraPhotoFlash" to cameraPhotoFlash,
            "cameraVideoTorch" to cameraVideoTorch
        )
    }
}
